besu/src/test/java/org/hyperledger/besu/util/EphemeryGenesisFileTest.java

```diff
@@ -0,0 +1,136 @@
+    assertThat(config.getTimestamp()).isZero();
+  }
+
+  @Test
+  public void testEphemeryWhenGenesisJsonIsInvalid() {
+    assertThatThrownBy(INVALID_GENESIS_JSON::getDifficulty)
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Invalid genesis block configuration");
+  }
+
+  @Test
+  public void testEphemeryWhenGenesisJsonIsValid() {
+    final GenesisConfigFile config = GenesisConfigFile.fromConfig(VALID_GENESIS_JSON.toString());
+    assertThat(String.valueOf(config.getTimestamp())).isNotNull();
+    assertThat(String.valueOf(config.getTimestamp())).isNotEmpty();
+    assertThat(String.valueOf(config.getConfigOptions().getChainId())).isNotNull();
+    assertThat(String.valueOf(config.getConfigOptions().getChainId())).isNotEmpty();
+  }
+
+  @Test
+  public void testEphemeryNotYetDueForUpdate() {
+    final GenesisConfigFile config = GenesisConfigFile.fromConfig(VALID_GENESIS_JSON.toString());
+    assertThat(CURRENT_TIMESTAMP).isLessThan(config.getTimestamp() + PERIOD_IN_SECONDS);
+  }
+
+  @Test
+  void testOverrideWithUpdatedChainIdAndTimeStamp() {
+    BigInteger expectedChainId =
+        BigInteger.valueOf(GENESIS_CONFIG_TEST_CHAINID)
+            .add(BigInteger.valueOf(PERIOD_SINCE_GENESIS));
+
+    long expectedGenesisTimestamp =
+        GENESIS_TEST_TIMESTAMP + (PERIOD_SINCE_GENESIS * PERIOD_IN_SECONDS);
+
+    final GenesisConfigFile config = GenesisConfigFile.fromResource("/ephemery.json");
+
+    final Map<String, String> override = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    override.put("chainId", String.valueOf(expectedChainId));
+    override.put("timestamp", String.valueOf(expectedGenesisTimestamp));
+
+    assertThat(config.withOverrides(override).getConfigOptions().getChainId()).isPresent();
+    assertThat(config.withOverrides(override).getTimestamp()).isNotNull();
+  }
+
+  @Test
+  public void testEphemeryWhenSuccessful() {
+    final GenesisConfigFile config = GenesisConfigFile.fromConfig(VALID_GENESIS_JSON.toString());
+
+    BigInteger expectedChainId =
+        BigInteger.valueOf(GENESIS_CONFIG_TEST_CHAINID)
+            .add(BigInteger.valueOf(PERIOD_SINCE_GENESIS));
+
+    long expectedGenesisTimestamp =
+        GENESIS_TEST_TIMESTAMP + (PERIOD_SINCE_GENESIS * PERIOD_IN_SECONDS);
+
+    EPHEMERY.setNetworkId(expectedChainId);
+
+    final Map<String, String> override = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    override.put("chainId", String.valueOf(expectedChainId));
+    override.put("timestamp", String.valueOf(expectedGenesisTimestamp));
+    config.withOverrides(override);
+
+    assertThat(CURRENT_TIMESTAMP_HIGHER)
+        .isGreaterThan(Long.parseLong(String.valueOf(GENESIS_TEST_TIMESTAMP + PERIOD_IN_SECONDS)));
+
+    assertThat(override.get("timestamp")).isEqualTo(String.valueOf(expectedGenesisTimestamp));
+    assertThat(override.get("chainId")).isEqualTo(expectedChainId.toString());
```

Comment:
this is just asserting that the desired values are in the override map. Can we also assert that the updatedConfig has these values applied